src/services/enums/reasons.ts

```diff
@@ -0,0 +1,42 @@
+import * as genericEnum from './generic';
+import db from '../../models';
+import { ENTITY_TYPE } from '../../constants';
+
+const { Reasons } = db;
+
+const findAll = async (
+  data: { validFor: typeof ENTITY_TYPE[keyof typeof ENTITY_TYPE] },
+) => genericEnum.findAll(Reasons, data);
```

Comment:
many of these functions could be made generic, IE, we only need a separate findAll for reasons and status if they are doing something besides passing a model through.

Reply:
Also, in this case, I believe the tablename should be Reason (singular)

Reply:
I have a generic implementation, these functions call it. They pass into it the "unique" thing each needs. How would you change it?

Reply:
```js
const findAll = async (
  model: ModelType, data: { validFor: typeof ENTITY_TYPE[keyof typeof ENTITY_TYPE] },
) => genericEnum.findAll(model, data);
```

etc

Reply:
discussed this on a call - clarified that what I'd really be after is no middleman function at all, IE we just call genericEnum.findAll where appropriate. Also stressed that it's a personal preference thing rather than a "make this change right now" kind of suggestion.